Index: trunk/configure.ac
===================================================================
--- trunk.orig/configure.ac
+++ trunk/configure.ac
@@ -186,7 +186,8 @@
 if test x"$enable_gtk2" = x"yes"; then
     # check for gdk2
     gtk2_binary_version=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
-    GTK2_IM_MODULEDIR="$libdir"/gtk-2.0/$gtk2_binary_version/immodules
+    gtk2_libdir=`$PKG_CONFIG --variable=libdir gtk+-2.0`
+    GTK2_IM_MODULEDIR="$gtk2_libdir"/gtk-2.0/$gtk2_binary_version/immodules
 
     PKG_CHECK_MODULES(GDK2, [
         gdk-2.0
Index: trunk/configure
===================================================================
--- trunk.orig/configure
+++ trunk/configure
@@ -15207,7 +15207,8 @@
 if test x"$enable_gtk2" = x"yes"; then
     # check for gdk2
     gtk2_binary_version=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
-    GTK2_IM_MODULEDIR="$libdir"/gtk-2.0/$gtk2_binary_version/immodules
+    gtk2_libdir=`$PKG_CONFIG --variable=libdir gtk+-2.0`
+    GTK2_IM_MODULEDIR="$gtk2_libdir"/gtk-2.0/$gtk2_binary_version/immodules
 
 
 pkg_failed=no
